src/rust/engine/options/src/parse_tests.rs

```diff
@@ -60,6 +60,22 @@ fn test_parse_quoted_string() {
     check_str("hanakapi'ai", r#""hanakapi'ai""#);
     check_str("bs\"d", r#""bs\"d""#);
     check_str("1995", r#""1995""#);
+    check_str("some\tembedded\nescapes\\", r"'some\tembedded\nescapes\\'");
+    check_str("non-escaping \\w backslash", r"'non-escaping \w backslash'");
+    check_str(
+        "some \u{0} octal \u{3f} values \u{53}",
+        r"'some \0 octal \77 values \123'",
```

Comment:
I think there might be a few extra things worth testing here:

- almost-octal escapes, e.g. `\9` -> `"\\9"`, `\468` -> `"&8"`
- larger than a `u8`: `\777` -> `\u{1ff}` (ǿ)

Either of these I could imagine a future refactoring getting slightly wrong, but the first one especially.